qa/tasks/cephfs/test_failover.py

```diff
@@ -706,6 +708,78 @@ def test_rank_stopped(self):
         status = self._confirm_single_replay()
         self.assertTrue(standby_count, len(list(status.get_standbys())))
 
+    def _teardown(self, open_proc):
+        '''
+        Teardown for test_health_warn_oversize_cache_has_no_counters().
+        '''
+        open_proc.stdin.close()
+        # XXX: run this beforehand because unmounting attempted during
+        # teardown will get stuck after "ceph fs fail --yes-i-really-mean-it"
+        # runs successfully.
+        self.mount_a.umount_wait()
+
+    def test_health_warn_oversize_cache_has_no_counters(self):
+        '''
+        Test that when MDS cache size crosses the limit, health warning
+        printed for standy-replay MDS doesn't include inode and stray
+        counters.
+
+        Tests: https://tracker.ceph.com/issues/63514
+        '''
+        # reduce MDS cache limit, default MDS cache limit is too high which
+        # will unnecessarily consume too many resources and too much time.
+        self.run_ceph_cmd('config set mds mds_cache_memory_limit 1K')
+        # health warning for crossing MDS cache size limit won't be raised
+        # until a threshold. default threshold is too high. it will
+        # unnecssarily consume so much time and resources.
+        self.run_ceph_cmd('config set mds mds_health_cache_threshold 1.000001')
+        # so that there is only active MDS and only 1 health warning is
+        # produced. presence of 2 warning should cause this test to fail
+        self.fs.set_max_mds(1)
+        self.run_ceph_cmd(f'fs set {self.fs.name} allow_standby_replay true')
+        self._confirm_single_replay()
+        self.fs.wait_for_daemons()
+        # The call above (to self.fs.wait_for_daemons()) should ensure we have
+        # only 1 active MDS on cluster
+        active_mds_id = self.fs.get_active_names()[0]
+        sr_mds_id = self.fs.get_standby_replay_names()[0]
+
+        # this should generate MDS cache more than 3M and therefore should
+        # trigger health warning MDS_CACHE_OVERSIZED.
+        open_proc = self.mount_a.open_n_background(".", 4000)
+
+        # actual test begins now...
+        errmsg = ('this test didn\'t generate health warning it was supposed '
+                  'to. health report has been logged above.')
+        with safe_while(sleep=3, tries=10, action=errmsg) as proceed:
+            check_now = False
+            while proceed():
+                cache_consumed = self.get_ceph_cmd_stdout(
+                    f'daemon mds.{active_mds_id} cache status')
+                log.info(f'cache_consumed = {cache_consumed}')
+
+                health_report = self.get_ceph_cmd_stdout('health detail')
```

Comment:
Isn't there a json format for this?

Reply:
True. Much better with it. Thanks! I've incorporated this.